IB_iterators.py

```diff
@@ -0,0 +1,77 @@
+import random
+
+# Task 1
+
+
+def gen_fa(path):
+    file = open(path)
+    lines = file.readlines()
+    print(lines)
+    file.close()
+    i = 0
+    while i < len(lines)-1:
+        yield lines[i].strip('\n') + ' ' + lines[i+1].strip('\n')
+        i = i + 2
+
+
+print('Please type path to example fasta file')
+path = input()
+
+print(gen_fa(path))
+for item in gen_fa(path):
+    print(item)
+
+print('Would you like to start eternal evolution? yes/no')
+answer = input()
+if answer == 'yes':
+    pass
+else:
+    raise KeyboardInterrupt
+
+# Task 2
+
+
+class Evolution:
+    def __init__(self, path):
+        self.path = path
+        file = open(self.path)
+        self.lines = file.readlines()
+        file.close()
+        self.seqs = []
+        while True:
+            for item in self.iter_gen():
+                self.seqs.append(item)
+            print(*self.seqs)
+
+    def deletion(self, it, ind):
+        it = it.replace(it[ind], "")
+        return it
+
+    def insertion(self, it, ind):
+        it_list = it.split()
+        insert = random.choice(it_list)
+        it = it[:ind] + insert + it[ind:]
+        return it
+
+    def replacement(self, it, ind):
+        it_list = it.split()
+        repl = random.choice(it_list)
+        it = it.replace(it[ind], repl)
+        return it
+
+    def mutate(self, it):
+        ind = random.randint(0, len(it)-1)
+        fun_list = ['self.deletion(it, ind)', 'self.insertion(it, ind)', 'self.replacement(it, ind)']
+        func = random.choice(fun_list)
+        return eval(func)
```

Comment:
`eval` это крайне небезопасная штука. Её лучше не использовать. Вот так будет лучше
```suggestion
        fun_list = [self.deletion, self.insertion, self.replacement]
        func = random.choice(fun_list)
        return func(it, ind)
```